src/kOS.Safe/Encapsulation/Structure.cs

```diff
@@ -26,7 +29,26 @@ static Structure()
         protected Structure()
         {
             instanceSuffixes = new Dictionary<string, ISuffix>(StringComparer.OrdinalIgnoreCase);
-            InitializeInstanceSuffixes();
+            RegisterInitializer(InitializeInstanceSuffixes);
+        }
+
+        protected void RegisterInitializer(Action callback)
+        {
+            if (!initializeSuffixCallbacks.Contains(callback))
+            {
+                initializeSuffixCallbacks.Add(callback);
+            }
+        }
+
+        private void callInitializeSuffixes()
+        {
+            if (needToInitializeSuffixes)
+            {
+                foreach (var callback in initializeSuffixCallbacks)
+                {
+                    callback.Invoke();
+                }
+            }
```

Comment:
Yeah, that will still work.  The lazy-load will just happen after the overridden `GetSuffix` method calls `base.GetSuffix` The only way this lazy load would fail is if the base method isn't called.